cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -269,6 +269,10 @@ public void scanApplications(Cursor cursor,
         StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD);
   }
 
+  public long getApplicationCount(String namespaceId) throws IOException {
```

Comment:
You actually never need one per namespace, but instead always query across all namespaces. So why not having one that query across all namespaces such that it can be done in one query?

Reply:
removed namespace query